File: lakeSide-hotel-demo-client-main/src/model/Surcharge.js
class Surcharge {
    constructor(id = null, content = null, quantity = null, price = null, 
        total = null, bookedRoom = null, status = null) {
        this.id = id; // Mặc định là null (kiểu bigint)
        this.content = content; // Chuỗi (String), mặc định là null
        this.quantity = quantity; // Số (number), mặc định là null
        this.price = price; // Số (number), mặc định là null
        this.total = total; // Số (number), mặc định là null
        this.bookedRoom = bookedRoom; // Đối tượng của class BookedRoom, mặc định là null
        this.status = status; // Chuỗi (String), mặc định là null
    }
}

export default Surcharge;

File: lakeSide-hotel-demo-server-master/src/main/java/com/dailycodework/lakesidehotel/model/Surcharge.java
package com.dailycodework.lakesidehotel.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Surcharge {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private  Long id;
    private String content;
    private int quantity;
    private BigDecimal price;
    private BigDecimal total;
    private String status;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "bookedRoom_id")
    private BookedRoom bookedRoom;

    public void SetTotal(){
        this.total = price.multiply(BigDecimal.valueOf(quantity));
    }
}
